File: ruby/prime-factors/prime_factors.rb
class PrimeFactors
  def self.for(num)
    prime_factors = []
    factor = 2

    while num > 1
      if (num % factor).zero?
        prime_factors << factor
        num /= factor
      else
        factor += 1
      end
    end

    prime_factors
  end
end

File: javascript/prime-factors/prime-factors.js
function primeFactors(input) {
  var num = input;
  var factors = [];
  var current_factor = 2;

  if (num < 2) { return factors };

  while (num > 1) {
    if (num % current_factor === 0) {
      factors.push(current_factor)
      num /= current_factor;
    } else {
      current_factor += 1;
    }
  }
  return factors;
};

module.exports = {for: primeFactors};
